### PR TITLE
CHANGELOG for 0.73.0 release

### PR DESCRIPTION
**Pull request by atoulme**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,15 @@
 
 ## Unreleased
 
+## v0.73.0
+
+This Splunk OpenTelemetry Collector release includes changes from the [opentelemetry-collector v0.73.0](https://github.com/open-telemetry/opentelemetry-collector/releases/tag/v0.73.0) and the [opentelemetry-collector-contrib v0.73.0](https://github.com/open-telemetry/opentelemetry-collector-contrib/releases/tag/v0.73.0) releases where appropriate.
+
+### 💡 Enhancements 💡
+- [Add java zc options to linux installer script](https://github.com/signalfx/splunk-otel-collector/pull/2718)
+- [Update installer script for generate_service_name option](https://github.com/signalfx/splunk-otel-collector/pull/2730)
```
Comment (aunshc) [inline CHANGELOG.md:11]:
```suggestion
- [Added profiling and JVM metrics options to linux installer script for zero configuration auto instrumentation of Java apps](https://github.com/signalfx/splunk-otel-collector/pull/2718)
- [Update installer script with option to generate service name with zero configuration auto instrumentation of Java apps](https://github.com/signalfx/splunk-otel-collector/pull/2730)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,15 @@
...
+- [Build experimental linux arm64 agent-bundle](https://github.com/signalfx/splunk-otel-collector/pull/2671)
+
```
Comment (aunshc) [inline CHANGELOG.md:13]: I believe we should add references to the addition of the zc options for Ansible, Puppet, Chef and Salt too.

File: CHANGELOG.md (hunk 1 above)
Comment (jeffreyc-splunk) [reply CHANGELOG.md:11]: I think we can remove the reference to #2730 since it was just to address an issue from #2718.

File: CHANGELOG.md (hunk 2 above)
Comment (jeffreyc-splunk) [reply CHANGELOG.md:13]: How about something like:
```suggestion
- Added profiling and JVM metrics options for zero configuration auto instrumentation of Java apps (Linux only):
  - [Installer script](https://github.com/signalfx/splunk-otel-collector/pull/2718)
  - [Ansible v0.16.0](https://github.com/signalfx/splunk-otel-collector/pull/2729)
  - [Chef v0.5.0](https://github.com/signalfx/splunk-otel-collector/pull/2733)
  - [Puppet v0.9.0](https://github.com/signalfx/splunk-otel-collector/pull/2734)
  - [Salt](https://github.com/signalfx/splunk-otel-collector/pull/2735)
```

Comment (aunshc) [reply CHANGELOG.md:13]: Nice, +1. Thanks Jeff!

File: CHANGELOG.md (hunk 1 above)
Comment (aunshc) [reply CHANGELOG.md:11]: Makes sense, we should update this to reflect this option as well then (Added profiling, JVM metrics and service name generation options for zero configuration auto instrumentation of Java apps).

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,20 @@
 
 ## Unreleased
 
+## v0.73.0
+
+This Splunk OpenTelemetry Collector release includes changes from the [opentelemetry-collector v0.73.0](https://github.com/open-telemetry/opentelemetry-collector/releases/tag/v0.73.0) and the [opentelemetry-collector-contrib v0.73.0](https://github.com/open-telemetry/opentelemetry-collector-contrib/releases/tag/v0.73.0) releases where appropriate.
+
+### 💡 Enhancements 💡
+- [Added profiling and JVM metrics options to linux installer script for zero configuration auto instrumentation of Java apps](https://github.com/signalfx/splunk-otel-collector/pull/2718)
+- [Update installer script with option to generate service name with zero configuration auto instrumentation of Java apps](https://github.com/signalfx/splunk-otel-collector/pull/2730)
```
Comment (jeffreyc-splunk) [inline CHANGELOG.md:11]: We can remove these 2 references to the installer script since they are included in the list below.

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,24 @@
...
+- [Build experimental linux arm64 agent-bundle](https://github.com/signalfx/splunk-otel-collector/pull/2671)
+- Added profiling and JVM metrics options for zero configuration auto instrumentation of Java apps (Linux only):
```
Comment (jeffreyc-splunk) [inline CHANGELOG.md:13]: Per @aunshc's suggestion: 
```suggestion
- Added profiling, JVM metrics, and service name generation options for zero configuration auto instrumentation of Java apps
```

File: CHANGELOG.md (hunk 3 above)
Comment (atoulme) [reply CHANGELOG.md]: done

**Pull request merged**